ja/core-libraries/email.rst

```diff
@@ -3,18 +3,527 @@ Email
+``helpers()`` メソッドを使うことで追加でヘルパーをロードできます。 ::
+
+    $email->helpers(['Html', 'Custom', 'Text']);
+
+ヘルパーを設定する時は、’Html’ を含めて下さい。そうしなければ、メールテンプレートにロードされません。
+
+もし、プラグインの中でテンプレートを使用してメール送信したい場合、おなじみの :term:`プラグイン記法`
+を使います。 ::
+
+    $email = new Email();
+    $email->template('Blog.new_comment', 'Blog.auto_message');
+
+上記の例は、 Blog プラグインのテンプレートを使用しています。
+
+いくつかのケースで、プラグインで用意されたデフォルトのテンプレートを上書きしたい場合があるかもしれません。
+``Email::theme()`` メソッドを使って適切なテーマを使用することを Email に伝えることによって行います。 ::
+
+    $email = new Email();
+    $email->template('Blog.new_comment', 'Blog.auto_message');
+    $email->theme('TestTheme');
+
+これは、Blog プラグインを更新せずにあなたのテーマの ``new_comment`` テンプレートで上書きできます。
+テンプレートファイルは、以下のパスで作成する必要があります:
+**src/Template/Plugin/TestTheme/Blog/Email/text/new_comment.ctp**
+
+添付ファイルの送信
+==================
+
+.. php:method:: attachments($attachments = null)
+
+メールにファイルを添付することができます。添付するファイルの種類や、
+宛先のメールクライアントにどのようなファイル名で送りたいのかによって幾つかの異なる書式があります。
+
+1. 文字列: ``$email->attachments('/full/file/path/file.png')`` は、
+   file.png というファイル名でこのファイルを添付します。
+2. 配列: ``$email->attachments(['/full/file/path/file.png'])`` は、
+   文字列の場合と同じ振る舞いをします。
+3. キー付き配列:
+   ``$email->attachments(['photo.png' => '/full/some_hash.png'])`` は、
+   photo.png というファイル名で some_hash.png ファイルを添付します。
+   受信者からは、some_hash.png ではなく photo.png として見えます。
+4. ネストした配列::
+
+    $email->attachments([
+        'photo.png' => [
+            'file' => '/full/some_hash.png',
+            'mimetype' => 'image/png',
+            'contentId' => 'my-unique-id'
+        ]
+    ]);
+
+   上記は、異なる mimetype と独自のコンテンツID を添付します
+   (添付をインラインに変換する場合にコンテンツIDをセットします)。
+   mimetype と contentId はこの形式のオプションです。
+
+   4.1. ``contentId`` を指定した時、HTML 内で ``<img src="cid:my-content-id">``
+   のようにファイルを使用できます。
+
+   4.2. 添付の ``Content-Disposition`` ヘッダーを無効にするために
+   ``contentDisposition`` オプションを使用できます。これは、outlook を使って
+   ical の招待状をクライアントに送る時に便利です。
+
+   4.3. ``file`` オプションの代わりに ``data`` オプションを使うと、
+   ファイル本文を文字列として添付することができます。これは、ファイルパスを指定せずに
+   添付することができます。
+
+トランスポートの使用
+====================
+
+トランスポートは、様々なプロトコルや方法でメールを送信するために設計されたクラスです。
+CakePHP は、 Mail (デフォルト)、 Debug と SMTP トランスポートをサポートします。
+
+これらの送信方法を設定するためには、 :php:meth:`Cake\\Mailer\\Email::transport()`
+メソッドを使用するか、設定内で transport を指定する必要があります。 ::
+
+    $email = new Email();
+
+    // Email::configTransport() を使ってすでに設定されたトランスポート名を使用
+    $email->transport('gmail');
+
+    // 構築されたオブジェクトを使用
+    $transport = new DebugTransport();
+    $email->transport($transport);
+
+独自のトランスポートの作成
+--------------------------
+
+SwiftMailer のような他のメールシステムを使うために独自のトランスポートを作成することができます。
+トランスポートを作るためには、(Example という名前のトランスポートの場合）最初に
+**src/Mailer/Transport/ExampleTransport.php** ファイルを作成してください。
+作成開始時点のファイルは次のようになります。 ::
+
+    namespace App\Mailer\Transport;
+
+    use Cake\Mailer\AbstractTransport;
+    use Cake\Mailer\Email;
+
+    class ExampleTransport extends AbstractTransport
+    {
+        public function send(Email $email)
+        {
+            // Magic inside!
```

Comment:
> Magic inside!

Should be translated. I don't know what it means though. I'll ask my team members. By the way, asking Dr. Google, I found [a nice song](https://youtu.be/buuYIBejsuM).